### DISCUSSION TITLE
TextX installation through CLI

### DISCUSSION THREAD

**Discussion by SalmanTayyab454** [textX/textX#341] [Q&A] [answered]
During the installation of textX it always throws an error of. "textX must be installed with CLI dependencies to use textx command.
And when I try the below command it throws an error that command not recognized.
pip install textX[cli].
How to fix this issue.Kindly guide. Thank you in advance.

**Comment by igordejanovic** [comment, marked as answer]
The message is just a warning that in order to use textX from command line you have to install [additional dependencies](https://github.com/textX/textX/blob/master/setup.cfg#L43). At this time it is just a `click` library.
Command not recognized suggest that you don't have `pip` properly installed in your environment.

**Comment by UffaModey** [comment]
I have this same problem, please. During my text installation, I get two exceptions thrown. "ModuleNotFoundError: No module named 'click'" and "textX must be installed with CLI dependencies to use". When I try to do the text installation with cli dependencies using "pip install textX[cli]" I get an error message saying "zsh: no matches found: textX[cli]". How do I add the click library and install textx with cli dependencies, please?

**Comment by igordejanovic** [comment]
zsh treats `[]` specially. Escape it with backslash:
```
pip install textx\[cli\]
```

**Comment by UffaModey** [comment]
Thank you very much. This worked. I am now able to run the textX command without errors from the cli.
